ops.qmd

```diff
@@ -2,84 +2,759 @@
+    * Infrastructure Management: Setting up and maintaining the infrastructure required for training and deploying models.
+    * Communication & Collaboration: Ensuring smooth communication and collaboration practices between data scientists, ML engineers, and operations teams.
+
+* Operational Excellence
+    * Monitoring: Techniques for monitoring model performance, data drift, and operational health.
+    * Governance: Implementing policies for model auditability, compliance, and ethical considerations.
+
+### Model Lifecycle Management
+
+![Figure 14.4: This diagram presents an overview of Model Lifecycle Management in an MLOps context, illustrating the flow from development to deployment and monitoring. The process begins with ML Development, where code and configurations are version-controlled. Data and model management are central to the process, involving datasets and feature repositories. Continuous training, model conversion, and model registry are key stages in the operationalization of training. Model deployment includes serving the model and managing serving logs. Alerting mechanisms are in place to flag issues, which feed into continuous monitoring to ensure model performance and reliability over time. This integrated approach ensures that models are not only developed but also maintained effectively throughout their lifecycle.](images/ai_ops/mlops_flow.png)
+
+#### Data Management
+
+In traditional centralized MLOps, data is aggregated into large datasets and data lakes, then processed on cloud or on-prem servers. However, embedded MLOps relies on decentralized data from local on-device sensors. Devices collect smaller batches of incremental data, often noisy and unstructured. With connectivity constraints, this data cannot always be instantly transmitted to the cloud and needs to be intelligently cached and processed at the edge.
+
+Embedded devices can only preprocess and clean data minimally before transmission due to limited on-device compute. Early filtering and processing occurs at edge gateways to reduce transmission loads. While leveraging cloud storage, more processing and storage happens at the edge to account for intermittent connectivity. Devices identify and transmit only the most critical subsets of data to the cloud.
+
+Labeling also faces challenges without centralized data access, requiring more automated techniques like federated learning where devices collaboratively label peers' data. With personal edge devices, data privacy and regulations are critical concerns. Data collection, transmission and storage must be secure and compliant.
+
+For instance, a smartwatch may collect step count, heart rate, GPS coordinates throughout the day. This data is cached locally and transmitted to an edge gateway when WiFi is available. The gateway processes and filters data before syncing relevant subsets with the cloud platform to retrain models.
```

Comment:
I don't totally know if this is in the scope of this chapter but I think it might add to the example to explain the process of the data coming off of the sensors on device, the signal processing that happens locally there, then the transmitting, and then the processing and filter that happens on the gateway and the cloud. Just good to give a full FOV of the data pipeline in the usecase you're using. Then you can specify what part of the overall process is the specific focus of mlops

Reply:
Hmmm. I like the idea. I will try to integrate this elsewhere. Otherwise, this might get cluttered.